include/ur_api.h

```diff
@@ -2739,6 +2765,41 @@ typedef enum ur_usm_alloc_info_t
 
 } ur_usm_alloc_info_t;
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief USM allocation descriptor type
+typedef struct ur_usm_desc_t
+{
+    ur_structure_type_t stype;                      ///< [in] type of this structure
+    const void* pNext;                              ///< [in][optional] pointer to extension-specific structure
+    ur_usm_mem_flags_t flags;                       ///< [in] memory allocation flags
+    ur_mem_advice_t hints;                          ///< [in] Memory advice hints
+    void* pool;                                     ///< [in][optional] Pointer to a pool created using urCreateUSMPool
```

Comment:
Yes, I wanted to do this but turns out we cannot have handle_t types inside the desc_t structs (see other comment). I think we could just make the pool a separate parameter as @jandres742 suggested.

Reply:
something like ur_usm_pool_handle_t?

Reply:
I mean, we can always change the script to remove that restriction ;) All we need is to make sure we have a forward declaration of the struct before we use it in here.

Reply:
Yeah, I think removing this restriction shouldn't be a big deal - we will still get validation during compilation. I was just wondering if there's some other reason for having this check.

Reply:
Ok. There was some discussion about this here: https://github.com/oneapi-src/unified-runtime/pull/302 I think it would be better to just have the pool handle as a separate argument. I've made this change as a separate commit. If this is fine I can squash it.